src/Pact/AST/Literals.hs

```diff
@@ -32,14 +32,29 @@ import GHC.Generics
 import Control.DeepSeq
 import Control.Lens
 
+import Data.ByteString (ByteString)
 import Data.Decimal
 import Data.Decimal.Orphans
 import Data.Hashable
 import Data.HashMap.Strict
+import Data.Set.NonEmpty (NESet)
 import Data.Text
 
 type Time = Double
 
+newtype KeySetPredicate = KeysetPredicate
+  { getPredicate :: Text
+  } deriving (Eq, Ord, Show, Generic, NFData)
+
+newtype PublicKey = PublicKey
+  { getPublicKey :: ByteString
+  } deriving (Eq, Ord, Show, Generic, NFData)
+
+data KeySet = KeySet
+  { ksPublicKeys :: !(NESet PublicKey)
```

Comment:
This is not something i'd care to make strict, simply because the set can be rather large.

Reply:
That's a good point.